src/store/modules/nft.js

```diff
@@ -493,12 +518,21 @@ const actions = {
     });
 
     const nftClassIds = Array.from(nftClassIdDataMap.keys());
+    const currentClassIds = [
+      ...new Set(
+        [...collectedNFTs, ...createdNFTClasses]
+          .map(item => item.class_id)
+          .filter(classId => classId !== undefined)
+      ),
+    ];
+    const mergedClassIds = [...new Set([...currentClassIds, ...nftClassIds])];
     await Promise.all(
-      nftClassIds.map(classId => {
+      mergedClassIds.map(classId => {
         const promises = [
           catchAxiosError(
             dispatch('populateNFTClassMetadataFromURIAndISCN', classId)
           ),
+          catchAxiosError(dispatch('fetchIscnRecords', classId)),
```

Comment:
A potential optimization is to first get the iscnIds from all nft classes, then deduplicate the iscnIds.
An iscnId can be parent of multiple nft classes

Reply:
I merged `fetchIscnRecords` into `populateNFTClassMetadataFromURIAndISCN`.